sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/changefeed/implementation/PartitionLoadBalancerImpl.java

```diff
@@ -60,64 +61,72 @@ public PartitionLoadBalancerImpl(
     public Mono<Void> start() {
         PartitionLoadBalancerImpl self = this;
 
-        return Mono.fromRunnable( () -> {
-            synchronized (lock) {
-                if (this.started) {
-                    throw new IllegalStateException("Partition load balancer already started");
-                }
-
-                this.started = true;
-                this.cancellationTokenSource = new CancellationTokenSource();
+        synchronized (lock) {
+            if (this.started) {
+                throw new IllegalStateException("Partition load balancer already started");
             }
 
-            CancellationToken cancellationToken = this.cancellationTokenSource.getToken();
+            this.started = true;
+            this.cancellationTokenSource = new CancellationTokenSource();
+        }
 
-            this.executorService.execute(() -> self.run(cancellationToken).block());
+        return Mono.fromRunnable( () -> {
+            Thread thread = new Thread(() -> self.run(self.cancellationTokenSource.getToken()).subscribe());
```

Comment:
why are we creating threads here?!

we should rely on what the reactor stream platform provide rather than creating our own threads.

Reply:
We create the corresponding workers for each partition range dynamically and they can go away based on changes to load balancing or splitting. Using reactor streams is an alternative that I'm still exploring outside of this PR.